### PR TITLE
function post commment has been added.

### PR DESCRIPTION
**Pull request by tomdud9907**

### CODE REVIEW

File: __tests__/app.test.js [hunk 1]
```diff
@@ -279,4 +279,73 @@ describe("Get /api/articles/:article_id/comments", () => {
...
+    }
+    return request(app)
+      .post("/api/articles/ljhbsdkjhb/comments")
+      .send(newComment)
+      .expect(400)
+      .then(({ body: { msg } }) => {
+        expect(msg).toBe("Invalid input")
+      })
+  })
+  test('should respond with a status 400 and "Invalid input" when passed a comment by an invalid username ', () => {
```
Comment (MrMcSnail) [inline __tests__/app.test.js:318]: should this be 404: username does not exist?

File: __tests__/app.test.js [hunk 2]
```diff
@@ -279,4 +279,73 @@ describe("Get /api/articles/:article_id/comments", () => {
...
+  test('should respond with a status 400 and "Invalid input" when passed comment is missing a body ', () => {
+    const newComment = { username: "butter_bridge" }
+    return request(app)
+      .post("/api/articles/2/comments")
+      .send(newComment)
+      .expect(400)
+      .then(({ body: { msg } }) => {
+        expect(msg).toBe("Invalid input")
+      })
+  })
```
Comment (MrMcSnail) [inline __tests__/app.test.js:350]: What should happen if the body has extra properties? Ignore them or error?

**Review by MrMcSnail — APPROVED** [review]
Good to merge

File: __tests__/app.test.js (hunk 1 above)
Comment (tomdud9907) [reply __tests__/app.test.js:318]: changed

**Pull request merged**